src/main/java/org/jabref/logic/cleanup/ConvertToBiblatexCleanup.java

```diff
@@ -39,8 +39,18 @@
                 entry.clearField(StandardField.YEAR).ifPresent(changes::add);
                 entry.clearField(StandardField.MONTH).ifPresent(changes::add);
             });
+        } else {
+            // If date field is filled and is equal to year it should be removed the year and month fields
+            entry.getFieldOrAlias(StandardField.DATE).ifPresent(date -> {
+                entry.getFieldOrAlias(StandardField.YEAR).ifPresent(checkYear -> {
+                    // Get the year of the date field (4 digits).
+                    if (date.substring(0, 4).equals(checkYear)) {
```

Comment:
Adding to @koppor's comment, we have a date class which you can use to parse the date in order to get the year AND the month.

Reply:
@koppor @tobiasdiez 
Please confirm that I understood correctly:
**Acceptance Criteria  1:**
- If the month + year fields are equal to the month + year in the date field, must clear the month and the year.
**Start:** 
date = "2020-05"
month = "05"
year = "2020"
**Result:** 
date = "2020-05"
month = ""
year = ""

**Acceptance Criteria 2:**
- If the month field is different from the month in the date field, does nothing
**Start:** 
date = "2020-05"
month = "04"
year = "2020"
**Result:** 
date = "2020-05"
month = "04"
year = "2020"

**Acceptance Criteria 3:**
- If the year field is different from the year in the date field, does nothing
**Start:** 
date = "2020-05"
month = "05"
year = "2019"
**Result:** 
date = "2020-05"
month = "05"
year = "2019"

**Acceptance Criteria 4:**
- If the date field has a blank/empty value, must set the date field with year and month values and clear the month + year fields.
**Start:** 
date = ""
month = "05"
year = "2020"
**Result:** 
date = "2020-05"
month = ""
year = ""

Thanks

Reply:
@koppor @tobiasdiez 
**Another question:**
What should happen if we have the following situation?
date = "2020"
month = "04"
year = "2020"
**Result:** ???

Reply:
I think in scenario 2 + 3 the correct information (the month in 2, the year in 3) can be deleted leaving only differing information behind.

Reply:
@tobiasdiez 
Just to make sure I got it right.
**Start:**
date = "2020-05"
month = "04"
year = "2020"

**Result:**
date = "2020-05"
month = "04"
year = ""

Please, confirms that the result is correct.

Reply:
For inconsistent data: Don't overwrite/move. Keep as is

Reply:
@tobiasdiez 
> Adding to @koppor's comment, we have a date class which you can use to parse the date in order to get the year AND the month.

What is the class name?

Reply:
https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/model/entry/Date.java